File: problem-solving/parsers/py/_08.py
def parser(f):
	n = int(f.readline())
	formulas = [ f.readline().strip() for _ in range(n) ]

	for s in formulas:
		for c in s:
			assert('a' <= c and c <= 'z')
	assert(f.readline() == "\n")

	m = int(f.readline())
	cert = set( f.readline().strip() for _ in range(m) )
	for s in cert:
		for c in s:
			assert('a' <= c and c <= 'z')
	assert(f.readline().strip() == "")

	return cert, (formulas,)

def verifier(cert, ans):
	return ans in cert

def error(cert, input, ans):
	print("Input:")
	print("formulas = " + str(input[0]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()

File: problem-solving/parsers/java/_08.java
import java.util.Scanner;
import java.util.Set;

public class _08 implements Parser {
    public Object[] parser(Scanner f) {
        int n = f.nextInt();
        f.nextLine();
        String[] formulas = Parser.parse_arr_string(f, n);


        assert(f.nextLine() == "");

        int c = f.nextInt();
        f.nextLine();
        Set<String> cert = Parser.parse_set_string(f, c);

        return Parser.ret_parser(cert, (Object)formulas);
    }

    public Object run(Object[] input) {
        return Solutions.chemicalConcoctions((String[])input[0]);
    }

    public boolean verifier(Object cert, Object ans) {
        return ((Set<String>)cert).contains((String)ans);
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "formulas = %s\n" +
            "\n" +
            "Expected:\n" +
            "%s\n" +
            "\n" +
            "Actual:\n" +
            "%s\n",
            Parser.print_arr((String[])input[0]),
            Parser.print_set((Set<String>)cert),
            (String)ans
        ));
    }

}
